### ISSUE TITLE
How to configure Tornado correctly in IDEA ?

### ISSUE THREAD

**Issue by Index-DiLing** [beehive-lab/TornadoVM#303]
After installation and compilation,I passed the test and it shows I made it
But when I try to use it in IDEA,it doesn't work correctly
There is my configuration
![JKNGRWYF}{XW_ 9F$KL5TEG](https://github.com/beehive-lab/TornadoVM/assets/72368764/1d061333-fff2-4f45-8b90-5931ce6227e5)
![FVZFI@H(XT@N~Y1}~3Q 2%N](https://github.com/beehive-lab/TornadoVM/assets/72368764/1eea726c-0484-411d-9ae3-bb6eea8e8c6f)
![NVX6ULH4ZSP_922G %CAPSG](https://github.com/beehive-lab/TornadoVM/assets/72368764/4e26f5f2-e3c7-4e63-a2f6-2d7f73513c93)
![7P9EJM{9IZ`GA@1C03I3XIB](https://github.com/beehive-lab/TornadoVM/assets/72368764/4fb01402-a643-4d85-a127-b827a4392167)
![F(ADMO9NN9I{4F GMU_8J](https://github.com/beehive-lab/TornadoVM/assets/72368764/876c3b7b-641e-4c24-b3a5-2f3ec1a49ad3)
![S4IX}SF$6B3~)DB@NKDW8VU](https://github.com/beehive-lab/TornadoVM/assets/72368764/28b48f8c-9adc-4c5f-baff-dde813012ba0)
(Code source example):
![5 AY%C7L3G)S(P%)4_H%VVO](https://github.com/beehive-lab/TornadoVM/assets/72368764/e8bab65d-512b-4817-afaa-6dfb156f4b22)
I should have followed the instructions and add the libs to my project
Acturally,I'm a beginner at this,So please forgive me for asking for help in this way

**Comment by jjfumero** [comment]
Having a quick look at your config files, it seems to be correct. The error is related to the OpenCL installation (as no driver is present). Did you install the OpenCL driver/runtime? We can give you some pointers to do so, which OS are you using? 

Note that you do not need the Eclipse code formatter to run TornadoVM, this is just for as to develop the core of TornadoVM.

**Comment by Index-DiLing** [comment]
Thanks!
My OS is Windows11,I think OpenCL drivers should be included in the video driver(It's probably worth mentioning that the graphics card I used was Intel A380,is it incompatible?) 
What particularly confused me was that I passed the test and no errors were reported...

**Comment by jjfumero** [comment]
Yes, ARC GPUs are compatible with TornadoVM. I also use an ARC 750 for Windows 11. Apart from the Intel GPU driver, you need Intel oneAPI to get OpenCL and Level Zero runtimes/APIs. 
The oneAPI base toolkit is enough for TornadoVM: https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html

**Comment by Index-DiLing** [comment]
Thanks again....
I download the oneAPI base toolkit except for the complier in it
But it doen't work......
If only OpenCL driver is missing, OpenCL should be included in the graphics card driver....right?
Is there anything else going wrong...?
I've been trying for a long time, and I'm getting frustrated.Your support is very helpful for me

**Comment by jjfumero** [comment]
No problem. GPU drivers/runtimes are always a bit difficult to install, especially on Linux. Usually, you need two things to make OpenCL/CUDA, oneAPI, etc running.

1) GPU drivers from the GPU vendor. This will install the low-level OS driver in your system. 
2) APIs and runtimes to be able to program the GPUs (this includes the OpenCL runtime, SPIR-V, etc.)

So, If you are running on ARC GPUs, you need
a) the ARC GPU Driver: https://www.intel.com/content/www/us/en/products/docs/discrete-gpus/arc/software/drivers.html 
b) the oneAPI toolkit:  https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html 

This should be enough to run any OpenCL. SPIR-V program, thus, should be enough for TornadoVM.

**Comment by stratika** [comment]
hi, last week @jjfumero published a [video](https://www.youtube.com/watch?v=F_0E-zFe2CU&ab_channel=JuanFumero) that is relevant to this thread.
